src/tcgwars/logic/impl/gen4/MajesticDawn.groovy

```diff
@@ -217,45 +221,52 @@ public enum MajesticDawn implements LogicCardInfo {
           resistance P, MINUS20
           pokePower "Darkness Shade", {
             text "Once during your turn, when you put Darkrai from your hand onto your Bench, you may choose 1 of the Defending Pokémon. That Pokémon is now Asleep."
-            actionA {
+            onActivate {
+              if(it==PLAY_FROM_HAND && my.deck && confirm("Use Darkness Shade?")) {
+                powerUsed()
+                apply ASLEEP
+              }
             }
           }
           move "Dark Slumber", {
             text "10 damage. At the end of your opponent’s next turn, the Defending Pokémon is now Asleep."
             energyCost D
             attackRequirement {}
             onAttack {
-              damage 0
+              delayed {
+                before BETWEEN_TURNS, {
+                  if(turnCount + 1 <= bg.turnCount){
+                    if(all.contains(pcs)){
+                      bc "Dark Slumber takes effect"
+                      apply ASLEEP
+                    }
+                  }
             }
           }
           move "Dark Resolve", {
             text "40 damage. If the Defending Pokémon is Asleep, remove 4 damage counters from Darkrai."
             energyCost D, D, C
-            attackRequirement {}
             onAttack {
-              damage 0
+              damage 40
+              if (defending.isSPC(ASLEEP)) {
+                heal 40, self
             }
           }
-
         };
       case DIALGA_4:
         return basic (this, hp:HP100, type:METAL, retreatCost:3) {
           weakness R, PLUS20
           resistance P, MINUS20
-          move "", {
-            text "If damage. "
-            energyCost M
-            attackRequirement {}
-            onAttack {
-              damage 0
-            }
-          }
+          //TODO: Implement Adamant Orb held item
           move "Time Shift", {
             text "Draw cards until you have 6 cards in your hand."
             energyCost M
-            attackRequirement {}
-            onAttack {
-              damage 0
+            attackRequirement{
+              assert my.hand.size() < 6
+              assert my.deck
```

Comment:
Could you add a message for these asserts?
For example:
`assert my.deck : "Your deck is empty"`